### ISSUE TITLE
1.0 Release

### ISSUE THREAD

**Issue by Datseris** [JuliaDynamics/DrWatson.jl#77]
This package has come a long way! It started by my wish that scientists shouldn't have to code the same thing over and over again. Then @JonasIsensee joined and made significant contributions and we released the first beta. Since then, I was extremely happy to see three more people doing (significant) contributions, @tamasgal @sebastianpech and @asinghvi17 . 

Its probably time to reflect and see what steps are required for a 1.0 release. I believe DrWatson is now used by enough people to have undergone the basic necessary testing. I'll write below the list that comes to my mind, but everyone (tagged here or not) should contribute their thoughts or wishes.

- [x] Update the Poster
- [x] ~Rethink #43~ . I am not so sure this was a good idea, as it tries to suggest structuring your project into a "Julia package form", which was never the intention. I am also not sure how many scientists write tests for their research projects? (I don't...)
- [x] Implement #36 ? It is not really breaking, and only additive, but in the end of the day it does change what `tagsave` will output.
- [x] Close #63 . I don't think we can do anything about it.
- [x] Resolve #78 .
- [x] Finish #81 .
- [x] Fix https://github.com/JuliaDynamics/DrWatson.jl/issues/82
---

After 1.0 release I'll make a 10-minutes video introduction to DrWatson and upload it to youtube.

**Comment by sebastianpech** [comment]
#43: One of the first things I did after moving my code to a DrWatson project was collecting all tests from the src/ folders in the test folder. What I like about that is that for the collection of somehow related "Packages" stored in src/ I can run all tests with the workflow in #67. However, as long as we don't have JuliaLang/Pkg.jl#1253 and without the workaround in #67, I think having the test folder by default is not necessary.

#63: We haven't really discussed the possible solutions to this. Given that it's not possible to tell if a float was generated with scientific notation or not I see the following possibilities:
   1. Rounding to significant digits if round to decimal ones gives 0.
   2. Rounding to M digits where M = max(N,minimum that assures that
   result is ≠ 0) (eg. N=4) 
3. Fully relying on string interpolation
I don't like the current behavior of just printing 0.0.

#36: Would the diff then be stored e.g in the bson file? Or are we talking about an extra file? In general I think it's a good idea. I don't have the discipline (or see the necessity) to check in every small change that I do that might not end up in my final code. So having this diff would actually help. However this raises further concerns that have been bothering me: 

1. was already discusses (on discourse I think) about how to do
   tagsave on files that do not support adding data, like images and
2. I'm currently using packages where files are saved inside a
   function which I can't access from the calling script, so it's
   actually impossible to use tagsave before saving, because saving is
   basically automated inside this function.

I wrote a prototype code that uses cassette to hook into the save process and git annex metadata to store additional information. However that's still very very alpha, not well portable and also quite likely out of scope of DrWatson. Nevertheless it would be nice to have your opinion on that, maybe someone has a proper solution for dealing with such cases.

**Comment by tamasgal** [comment]
Really looking forward to the major release, I like this package a lot :wink: 

I have one thing to add to the list: albeit not having used the `produce_or_load` feature a lot yet, I have the feeling that I have a few ideas related to that. I will try to summarise them in near future!

**Comment by Datseris** [comment]
Hello, 

> I wrote a prototype code that uses cassette to hook into the save process and git annex metadata to store additional information. However that's still very very alpha, not well portable and also quite likely out of scope of DrWatson. Nevertheless it would be nice to have your opinion on that, maybe someone has a proper solution for dealing with such cases.

Sure, this seems useful! Let me know when you have this somewhere online!

> I have one thing to add to the list: albeit not having used the produce_or_load feature a lot yet, I have the feeling that I have a few ideas related to that. I will try to summarise them in near future!

Alright, I am looking forward to it!

--- 

Given the discussion here, I think we can neglect #63 completely and we can leave #43 as-is; having one extra folder is probably not too bad. The only thing I'd like to have is #36 , as it does change the output of `tagsave`. @JonasIsensee  any other comments?

**Comment by Datseris** [comment]
Just noticed we should resolve #78 as well before 1.0.

**Comment by Datseris** [comment]
Another thing: now the poster has space for one more example demonstration. Do you have any preference on what should I show there?

at the moment there are three examples showcasing :

* `savename`
* project navigation
* `collect_results`

**Comment by JonasIsensee** [comment]
Hey,
sorry for the late reply.
I don't have any more suggestions, right now I am happy with DrWatson.jl as it is.

**Comment by tamasgal** [comment]
> Another thing: now the poster has space for one more example demonstration. Do you have any preference on what should I show there?
> 
> at the moment there are three examples showcasing :
> 
> * `savename`
> * project navigation
> * `collect_results`

I'd love to see a tiny but "to the point" workflow of `produce_or_load`, but I am still too much of a novice with that feature to come up with a good example ;)

**Comment by Datseris** [comment]
Alright, here is the poster:

![DrWatson-poster-1 0 0](https://user-images.githubusercontent.com/19669089/63874828-885ac800-c9c2-11e9-9c21-901cc18f9bd2.jpg)

I coudln't really come up with a sufficiently small example of `produce_or_load`, so I thought of being more explicit with the `tag!` stuff, since we had some nice improvements there as well.

**Comment by Datseris** [comment]
Damnit, https://github.com/JuliaDynamics/DrWatson.jl/issues/82 came up, which must be resolved before the release. I took a first glance but didn't really know how to proceed (actually I am still in 1.1, waiting to go to 1.3 directly). If someone is already at 1.2. please have a look. I'll also have a look over the weekend.

**Comment by sebastianpech** [comment]
About the poster. Looks good :) Two things:
- What do you mean by "Safe-saving"? and
- I would use `datadir()` or even better `datadir("sims")` in the `collect_results` example to show the usage of the directory functions

**Comment by Datseris** [comment]
THanks, took care of all the comments:

![DrWatson-poster-1 0 0](https://user-images.githubusercontent.com/19669089/64028283-a4857300-cb42-11e9-8b39-1b5aaec0c3af.jpg)

**Comment by Datseris** [comment]
okay, engines set, go. I'll write an announcement post a bit later today and tag 1.0!
